stable/database/templates/service-clusterip.yaml

```diff
@@ -13,6 +14,33 @@ metadata:
     chart: {{ template "database.chart" . }}
     release: {{ .Release.Name | quote }}
   name: {{ .Values.database.name }}-{{ default .Values.admin.serviceSuffix.clusterip .Values.database.serviceSuffix.clusterip }}
+spec:
+  ports:
+  - { name: 48006-tcp,  port: 48006,  protocol: TCP,  targetPort: 48006 }
+  selector:
+    domain: {{ .Values.admin.domain }}
+    database: {{ .Values.database.name }}
+    component: te
```

Comment:
How do we prevent this Service from selecting the secondary releases? Is the release name present in the `database` label?

Reply:
This service (`demo-clusterip`) is selecting all TEs in a single cluster for database `demo`. This is for backwards compatibility.
I added another service (`group1-nuodb-cluster0-demo-database-clusterip`) which selects only TEs from a specific Helm release (TE group).

Reply:
It looks like the Service filters on `app=<database.fullname>` in the secondary release. But don't we also want to use that filter in the primary release, otherwise this Service will select the secondary TEs as well?

Reply:
So this is used by `direct=true` currently? Would it be breaking backwards compatibility to make this Service only select the TEs in the primary release? Maybe it would break what @acabrele is doing, but until we properly introduce this feature, we don't expect TEs to be spread across multiple Deployments (ignoring multi-cluster).

Which Service would select only the primary TEs?

Reply:
Your comment is on the cluster service installed only by the primary release so I'm confused a bit.

By design the following services should be all services deployed for TE group `group1` (primary) and `group2` (secondary):
* `demo-clusterip` selects all TEs from all TE groups (this is kept for backward compatibility)
* `group1-nuodb-cluster0-demo-database-clusterip` selects only TEs for `group1`
* `group2-nuodb-cluster0-demo-database-clusterip` selects only TEs for `group2`

Please let me know if the implementation doesn't match this somewhere.

Reply:
Okay, sorry. The second Service in this file is unconditionally created, so there would be a service specifically for the primary.